core/src/main/scala/spinal/core/AFix.scala

```diff
@@ -1025,37 +1041,43 @@ class AFix(val maxRaw: BigInt, val minRaw: BigInt, val exp: Int) extends MultiDa
+  private def getTrunc: TagAFixTruncated = {
+    this.getTag(classOf[TagAFixTruncated]).getOrElse(AFixTruncatedScope.get)
+  }
 
-  def round(exp: Int, aligned: Boolean): AFix = {
-    val trunc = this.getTag(classOf[TagAFixTruncated])
+  def round(exp: Int, aligned: Boolean = getTrunc.saturation): AFix = {
+    this._round(getTrunc.rounding, exp, aligned)
+  }
 
-    if (trunc.isDefined) {
-      this._round(trunc.get.rounding)
-    } else {
-      roundHalfToInf(exp)
-    }
+  def fixTo(af: AFix, roundType: RoundType): AFix = this._round(roundType, af.exp).sat(af)
+  def fixTo(af: AFix): AFix = this.fixTo(af, getTrunc.rounding)
+  def fixTo(Q: QFormat): AFix = {
+    val res = AFix(Q.width-Q.fraction exp, -Q.fraction exp, Q.signed)
+    res := this.fixTo(res)
+    res
   }
 
-  override def toString: String = s"${component.getPath() + "/" + this.getDisplayName()} : ${getClass.getSimpleName}[max=${maxRaw}, min=${minRaw}, exp=${exp}, bits=${raw.getWidth}]"
+  override def toString: String = {
+    s"${component.getPath() + "/" + this.getDisplayName()} : ${getClass.getSimpleName}[max=${maxRaw}($maxValue), min=${minRaw}($minValue), exp=${exp}($step), bits=${raw.getWidth}]"
+  }
 
-  private def _round(roundType: RoundType): AFix = {
+  private def _round(roundType: RoundType, exp: Int = 0, align: Boolean = getTrunc.saturation): AFix = {
     roundType match {
-      case RoundType.FLOOR       => this.floor(0)
-      case RoundType.CEIL        => this.ceil(0)
-      case RoundType.FLOORTOZERO => this.floorToZero(0)
-      case RoundType.CEILTOINF   => this.ceilToInf(0)
-      case RoundType.ROUNDUP     => this.roundHalfUp(0)
-      case RoundType.ROUNDDOWN   => this.roundHalfDown(0)
-      case RoundType.ROUNDTOZERO => this.roundHalfToZero(0)
-      case RoundType.ROUNDTOINF  => this.roundHalfToInf(0)
-      case RoundType.ROUNDTOEVEN => this.roundHalfToEven(0)
-      case RoundType.ROUNDTOODD  => this.roundHalfToOdd(0)
-      case RoundType.SCRAP        => this.scrap(0)
+      case RoundType.FLOOR       => this.floor(exp)
+      case RoundType.CEIL        => this.ceil(exp, align)
+      case RoundType.FLOORTOZERO => this.floorToZero(exp)
+      case RoundType.CEILTOINF   => this.ceilToInf(exp, align)
+      case RoundType.ROUNDUP     => this.roundUp(exp, align)
+      case RoundType.ROUNDDOWN   => this.roundDown(exp, align)
+      case RoundType.ROUNDTOZERO => this.roundToZero(exp, align)
+      case RoundType.ROUNDTOINF  => this.roundToInf(exp, align)
+      case RoundType.ROUNDTOEVEN => this.roundToEven(exp, align)
+      case RoundType.ROUNDTOODD  => this.roundToOdd(exp, align)
+      case RoundType.SCRAP        => this.scrap(exp)
     }
   }
 
-  def saturated(): AFix = this.truncated(saturation = true, overflow = false)
+  def saturated(rounding  : RoundType = RoundType.FLOOR): AFix = this.truncated(saturation = true, overflow = false, rounding = rounding)
 
   def truncated(saturation: Boolean = false,
```

Comment:
If we provide AFixTruncatedScope as a default, i guess the default value of that function should fetch AFixTruncatedScope current value ?

Reply:
hmm, seems like it is trying to do too many thing at one.

Why the saturated function would have rounding ?
I mean saturation is a specific thing.

def truncated is maybe missnamed, but is meant to be the function with all options.
Maybe we should rename def truncated into def reformated ?

Reply:
"truncated" can be regarded as truncating both LSB and/or MSB, I think? `rounded` and `saturated` are just two shortcuts of truncation with different settings.